tests/e2e/not-monetized.spec.ts

```diff
@@ -1,133 +1,124 @@
-  await expect(warning).toHaveText(msg);
-
-  await context.newPage().then((page) => page.bringToFront());
-  await expect(warning).toBeVisible();
-  await expect(warning).toHaveText(msg);
-});
-
-test('shows not monetized on internal pages', async ({
-  page,
+// We make extensive use of test.step in this test to reuse the same browser
+// instance, as connecting wallet each time for small tests is expensive.
+test('shows not-monetized status', async ({
   popup,
   i18n,
   channel,
+  background,
+  persistentContext: context,
 }) => {
-  const url = channel === 'msedge' ? 'edge://settings' : 'chrome://settings';
-  await page.goto(url);
-
   const warning = popup.getByTestId('not-monetized-message');
-  await expect(warning).toBeVisible();
-  await expect(warning).toHaveText(
-    i18n.getMessage('notMonetized_text_internalPage'),
-  );
-});
-
-test.describe('shows not monetized on non-https pages', () => {
-  let warning: Locator;
-  let msg: string;
-
-  test.beforeAll(async ({ popup, i18n }) => {
-    warning = popup.getByTestId('not-monetized-message');
-    msg = i18n.getMessage('notMonetized_text_unsupportedScheme');
-  });
-
-  test.beforeEach(async ({ popup, page }) => {
+  const msg = {
+    newTab: i18n.getMessage('notMonetized_text_newTab'),
+    internalPage: i18n.getMessage('notMonetized_text_internalPage'),
+    unsupportedScheme: i18n.getMessage('notMonetized_text_unsupportedScheme'),
+    noLinks: i18n.getMessage('notMonetized_text_noLinks'),
+  };
+
+  const newPage = async () => {
+    const page = await context.newPage();
     await popup.reload();
     await page.bringToFront();
-  });
-
-  test('http:// URLs', async ({ page }) => {
-    await page.goto('http://example.com');
+    return {
+      goto: page.goto.bind(page), // so we don't have to call `page.page.goto()` every time
+      page: page,
+      [Symbol.asyncDispose]: async () => {
+        await page.close();
+      },
+    };
+  };
+
+  await test.step('shows not monetized on empty tabs', async () => {
     await expect(warning).toBeVisible();
-    await expect(warning).toHaveText(msg);
-  });
-
-  test('navigating from https:// to http://', async ({ page, i18n }) => {
-    await page.goto('https://example.com');
-    await expect(warning).toBeVisible();
-    await expect(warning).toHaveText(
-      i18n.getMessage('notMonetized_text_noLinks'),
-    );
+    await expect(warning).toHaveText(msg.newTab);
 
-    await page.goto('http://example.com');
+    await using _page = await newPage();
     await expect(warning).toBeVisible();
-    await expect(warning).toHaveText(msg);
+    await expect(warning).toHaveText(msg.newTab);
   });
 
-  test('file:// URLs', async ({ page }) => {
-    await page.goto(pathToFileURL('.').href);
-    await expect(warning).toBeVisible();
-    await expect(warning).toHaveText(msg);
-  });
+  await test.step('shows not monetized on internal pages', async () => {
+    await using page = await newPage();
```

Comment:
This along with `Symbol.asyncDispose` will automatically close the page at then end of this code block.

We also reload popup on each new page create - to simulate new tab closing the popup in browser.